### ISSUE TITLE
Small Refactor of Subscription Builder's create() Method

### ISSUE THREAD

**Issue by frankieeedeee** [laravel/cashier-stripe#1122]
**TLDR;** Is it possible to extract the model creation operations in `Laravel\Cashier\SubscriptionBuilder::create()` into separate, protected methods, for better extensibility?

<hr>

It's already extremely easy to extend Cashier in a lot of various ways, by both consuming applications and extending packages. However one point where extensibility is a little more restricted is with the `create()` method of the Subscription Builder class.

### Example 
Consider a (hypothetical) use case where an application wanted to track the name of a Stripe Product that's associated with a billable's subscription item, inside of the `subscription_items` table. This would only require a single additional line inside of `SubscriptionBuilder::create()`...

````php
// Laravel\Cashier\SubscriptionBuilder, L329

foreach ($stripeSubscription->items as $item) {
    $subscription->items()->create([
        'stripe_id' => $item->id,
        'stripe_plan' => $item->plan->id,
        'quantity' => $item->quantity,
        'stripe_product_name' => $item->price->product->name, // Additional Line, could really be anything
    ]);
}
````

So, the consuming application could create a custom Subscription Builder class, which simply extends Cashier's Subscription Builder.

**However**, since the model creation logic is embedded directly in the `create` method, that custom class would have to re-implement (copy and paste) **all** of `Laravel\Cashier\Subscription`, just for a small amount of additional logic

````php
use Laravel\Cashier\SubscriptionBuilder;

class MyCustomSubscriptionBuilder extends SubscriptionBuilder
{
    public function create()
    {
        // ... Copy & paste of initial SubscriptionBuilder::create()

        foreach ($stripeSubscription->items as $item) {
            $subscription->items()->create([
                // Custom model logic / additional data
            ]);
        }

        // ... Copy & paste of remaining SubscriptionBuilder::create()
    }
}
````

... not ideal!

### Proposal

Could the model creation logic of `Laravel\Cashier\SubscriptionBuilder::create()` be extracted into separate, protected methods, to enable better extensibility?

I.e:
- `# Laravel\Cashier\SubscriptionBuilder::createSubscriptionModel(...)` and
- `# Laravel\Cashier\SubscriptionBuilder::createSubscriptionItemModel(...)`

With this, extending applications could simply override the relevant model creation method(s), whilst the create() method primarily focuses on the Stripe interaction.

Complete example not provided for the sake of reducing length of this issue, however more than happy to provide one! Additionally, would be happy to submit a PR for this, if this small refactoring is considered acceptable. Let me know if any more information is needed. Many thanks for all your work on such a great package! 🍻

**Comment by driesvints** [comment]
Thanks for the suggestion. I've implemented a new `createSubscription` method through this PR: https://github.com/laravel/cashier-stripe/pull/1124. I'm just going to leave it at that one method however and not add a separate one for the items.

**Comment by frankieeedeee** [comment]
Thanks for your reply @driesvints and for the quick PR + merge! 🙌 Makes complete sense re the single method. Cheers!
